src/server/scripts/Northrend/zone_zuldrak.cpp

```diff
@@ -948,6 +950,39 @@ class npc_storm_cloud : public CreatureScript
     }
 };
 
+class spell_scourge_disguise_instability : public AuraScript
+{
+    PrepareAuraScript(spell_scourge_disguise_instability)
+
+    void HandleApply(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+    {
+        SetDuration(urand(3*60*1000, 5*60*1000));
+    }
+
+    void HandleRemove(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+    {
+        Unit* caster = GetCaster();
+        if (caster && caster->IsPlayer())
+        {
+            Player* player = caster->ToPlayer();
+            if (player->HasAura(SPELL_SCOURGE_DISGUISE) || player->HasAura(SPELL_SCOURGE_DISGUISE_INSTANT_CAST))
+            {
+                WorldPacket data;
+                ChatHandler::BuildChatPacket(data, CHAT_MSG_RAID_BOSS_EMOTE, LANG_UNIVERSAL, player, player, "Scourge Disguise Failing! Find a safeplace!");
```

Comment:
Thanks for the PR but we don't hardcode messages like this. They are picked up from the DB

Reply:
I'm kinda new here... would you care to share a specific table to look?

Reply:
Well actually we ideally dont use chathandler here. 

What you want to do is
```cpp
player->Say(TEXT_ID);
```
text_id is from broadcast_text table

Reply:
and remember to not hardcode the textid, put it in an enum

Reply:
```sql
SELECT * FROM broadcast_text WHERE maletext LIKE '%Scourge Disguise%'

Reply:
Wow, so it is actually not 1, but 3 different texts. Probably should be used randomly.

Reply:
@Kitzunu I've replaced hardcoded string with TEXT_ID from Enum.
But I would like to use player->Talk instead of player->Say, because I need to set specific message type. This message should not be a regular chat message from player, but a big yellow text in the center of the screen:

![image](https://github.com/user-attachments/assets/0c889e17-6ee4-4756-a307-53c26ae91572)
Video from 2008: https://www.wowhead.com/wotlk/quest=12669/so-far-so-bad#videos:id=944

Reply:
Talk isn't a member of Player class

Reply:
So the correct way of fixing it is to know how it should be sent which can be obtained in a sniff.
That way we can figure who is the sender, the server or the player.

If it is the player, then we currently don't have a supported way.
But assuming it isn't you can do
```cpp
ChatHandler(player->GetSession()).PSendSysMessage(TEXT_ID);
```

Reply:
Remember to include ChatHandler.h

Reply:
and even if Talk was a part of Player class it doesn't do notifications

Reply:
So maybe acore_string and PSendSysMessage is the way to go...

@Nyeriah Assist! I am getting brain farts

Reply:
> Talk isn't a member of Player class

But it is a member of Unit class, which is a parent class for Player.
Is it "bad manners" to use parent class methods?

Reply:
> and even if Talk was a part of Player class it doesn't do notifications

I have ~0 knowledge of code base at this point, so I do not know what you mean by telling "it doesn't do notifications" :)

BUT! With my current implementation I have result pretty close to what Blizzard have in their serevers
![image](https://github.com/user-attachments/assets/b80bb597-0b4b-466b-b2cc-27937aa9b865)

Reply:
my bad then